src/Dialect/Krnl/KrnlOps.cpp

```diff
@@ -443,7 +444,16 @@ ParseResult KrnlIterateOp::parse(OpAsmParser &parser, OperationState &result) {
   Region *region = result.addRegion();
   SmallVector<Type, 4> inductionVarTypes(
       inductionVarRefs.size(), builder.getIndexType());
-  if (parser.parseRegion(*region, inductionVarRefs, inductionVarTypes))
+
+  SmallVector<OpAsmParser::Argument> entryArgs;
+  for (auto it : llvm::zip(inductionVarRefs, inductionVarTypes)) {
+    OpAsmParser::Argument arg;
+    arg.ssaName = std::get<0>(it);
+    arg.type = std::get<1>(it);
+    entryArgs.push_back(arg);
+  }
+
```

Comment:
It all look fairly straightforward except for this sequence. Do you mind just explaining this section? Is it something that used to be done inside the `parseRegion`? tx

Reply:
This is in response to this change: https://reviews.llvm.org/D124649. I can see if there's a neater way to make the change, but this was pretty straightforward without impacting the rest of the code

Reply:
Code is ok as long as it works functionally. Thanks for the reference, will be useful in the future if we need to revisit or apply elsewhere.